### PR TITLE
Add development guideline and docs

### LINKED ISSUE #1556: Add Development guideline
Scope: Make docs about Otter internals

What Docs should help with:
- Starting to work in Otter.
- Project structure.
- How big feature works (Dynamic Content & Condition, Form Block, etc)
- How to create blocks. 
- Where to go in case of problems.

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Closes #1556 .

### Summary

This PR will include docs about essential information for people non-familiar with the Otter codebase.
- How to make a running WP instance with Otter locally.
- Project structure, where to look for things.
- Details about different plugins and how they work.
- Handbook with general tips and tricks.
- Testing checklist & cheat sheet.
- PR & Coding best practices.

### CODE REVIEW

**Comment by HardeepAsrani** [comment]
Looks good, it can have a section about the feature-plugin, as in Blocks CSS/Animation and so on.

File: docs/woocommerce-extensions.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# WooCommerce Extensions
+
+In WordPress, WooCommerce is the most popular plugin for e-commerce. In a simple view of a plugin developer: people with business that have a high chance to pay for a premium plugin. The market is so big that we can't ignore it. Otter can not a respectable plugin, without WooCommerce integration.
+
+The available WooCommerce extensions are:
+- Add to cart Block
+- Woo Comparison Block
```
Comment (HardeepAsrani) [inline docs/woocommerce-extensions.md:7]: We don't have it anymore, it's part of Sparks now.

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
Apart from that, few things we can mention:

- We can mention CSS Utility class.
- We can mention how to add new patterns.

**Review by arinaturcu — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**